nifi-nar-bundles/nifi-lumberjack-bundle/nifi-lumberjack-processors/src/main/java/org/apache/nifi/processors/lumberjack/ListenLumberjack.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+
+    @Override
+    protected String getBatchKey(LumberjackEvent event) {
+        return event.getSender();
+    }
+
+    protected void respond(final LumberjackEvent event, final LumberjackResponse lumberjackResponse) {
+        final ChannelResponse response = new LumberjackChannelResponse(lumberjackEncoder, lumberjackResponse);
+
+        final ChannelResponder responder = event.getResponder();
+        responder.addResponse(response);
+        try {
+            responder.respond();
+        } catch (IOException e) {
+            getLogger().error("Error sending response for transaction {} due to {}",
+                    new Object[] {event.getSeqNumber(), e.getMessage()}, e);
+        }
+    }
+
+    protected void postProcess(final ProcessContext context, final ProcessSession session, final List<LumberjackEvent> events) {
+        // first commit the session so we guarantee we have all the events successfully
+        // written to FlowFiles and transferred to the success relationship
+        session.commit();
+        // respond to each event to acknowledge successful receipt
+        for (final LumberjackEvent event : events) {
+        respond(event, LumberjackResponse.ok(event.getSeqNumber()));
+        }
+    }
+
+    @Override
+    protected String getTransitUri(FlowFileEventBatch batch) {
+        final String sender = batch.getEvents().get(0).getSender();
+        final String senderHost = sender.startsWith("/") && sender.length() > 1 ? sender.substring(1) : sender;
+        final String transitUri = new StringBuilder().append("lumberjack").append("://").append(senderHost).append(":")
+                    .append(port).toString();
+    return transitUri;
+    }
+
+    @Override
+    protected Map<String, String> getAttributes(FlowFileEventBatch batch) {
+    final List<LumberjackEvent> events = batch.getEvents();
+    // the sender and command will be the same for all events based on the batch key
+    final String sender = events.get(0).getSender();
+    final int numAttributes = events.size() == 1 ? 5 : 4;
+    final Map<String,String> attributes = new HashMap<>(numAttributes);
+    attributes.put(LumberjackAttributes.SENDER.key(), sender);
+    attributes.put(LumberjackAttributes.PORT.key(), String.valueOf(port));
+    attributes.put(CoreAttributes.MIME_TYPE.key(), "text/plain");
+    // if there was only one event then we can pass on the transaction
+    // NOTE: we could pass on all the transaction ids joined together
+    if (events.size() == 1) {
+        attributes.put(LumberjackAttributes.SEQNUMBER.key(), String.valueOf(events.get(0).getSeqNumber()));
+
+        // Convert the serialezed fields from JSON
+        String serialFields = String.valueOf(events.get(0).getFields());
+        Gson jsonObject = new Gson();
+
+        Map<String, String> fields = jsonObject.fromJson(serialFields, Map.class);
+
+        // TODO: Allow user to select what fields get added
```

Comment:
Did you want to apply this as part of the process or defer?  An UpdateAttribute could take care of this after bringing the data into NiFi, but will defer to you.